include/seqan3/core/type_traits/iterator.hpp

```diff
@@ -148,89 +148,88 @@ struct size_type<it_t>
 #endif // SEQAN3_DEPRECATED_310
 
 // see specialisation for ranges in core/type_traits/range.hpp
+//!\}
 
-// ----------------------------------------------------------------------------
-// iterator_tag
-// ----------------------------------------------------------------------------
+} // namespace seqan3
+
+namespace seqan3::detail
+{
+#if SEQAN3_WORKAROUND_GCC_96070
+//!\cond
+template <typename it_t>
+struct iterator_category_tag
+{
+    using iterator_category = void;
+};
 
+template <typename it_t>
+    requires requires { typename std::iterator_traits<it_t>::iterator_category; }
+struct iterator_category_tag<it_t>
+{
+    using iterator_category = typename std::iterator_traits<it_t>::iterator_category;
```

Comment:
```suggestion
    using type = typename std::iterator_traits<it_t>::iterator_category;
```
? Otherwise it is not a seqan3::transformation_trait

Reply:
Does it need to be a `transformation_trait`, though? I've no strong feelings, so I'll change it :)